src/main/java/net/rptools/maptool/client/functions/MapFunctions.java

```diff
@@ -397,6 +399,29 @@ public Object childEvaluate(
       return (MapTool.getFrame().getToolbarPanel().getMapselect().isVisible()
           ? BigDecimal.ONE
           : BigDecimal.ZERO);
+    } else if ("setMapVision".equalsIgnoreCase(functionName)) {
+      FunctionUtil.blockUntrustedMacro(functionName);
+      FunctionUtil.checkNumberParam(functionName, parameters, 1, 1);
+      Zone currentZR = MapTool.getFrame().getCurrentZoneRenderer().getZone();
+      if (parameters.get(0).toString().equalsIgnoreCase("off")) {
```

Comment:
You will need to deal with the case where the string is not valid (e.g. not off/day/night) with an else 

You can also do this with case, 
```java
switch(parameters.get(0).toString().toLowerCase()) {
     case "off" -> MapTool.serverCommand().setVisionType(currentZR.getId(), Zone.VisionType.OFF);
     case "day" -> MapTool.serverCommand().setVisionType(currentZR.getId(), Zone.VisionType.DAY);
     case "night" -> MapTool.serverCommand().setVisionType(currentZR.getId(), Zone.VisionType.NIGHT);
     default ->  throw new ParserException(...);
}